cli/js/lib.deno.ns.d.ts

```diff
@@ -543,6 +537,24 @@ declare namespace Deno {
     /** Read-write. Behaves like `x` and allows to read from file. */
     | "x+";
 
+  // @url js/tty.d.ts
+
+  /** Check if a given resource is TTY. */
+  export function isatty(rid: number): boolean;
+  /** Extended file abstraction for TTY input */
+  export class TTYInput extends File {
+    /** Is TTY under raw mode. */
+    get isRaw(): boolean;
+    /** Set TTY to be under raw mode. */
+    setRaw(mode: boolean): void;
+  }
+  /** An instance of `TTYInput` for stdin. */
+  export const stdin: TTYInput;
+  /** An instance of `File` for stdout. */
+  export const stdout: File;
+  /** An instance of `File` for stderr. */
+  export const stderr: File;
```

Comment:
I like the minimality of `isatty(rid)`. Maybe you can do something similar here, instead of defining this new class `TTYInput` you could just add a new function... something like `Deno.setRaw(rid)`. What do you think?

Additionally can you mark in the JSDOC any new APIs as 
```
/** UNSTABLE: newly added API
  *
  *  Check if a given resource is TTY
  */
export function isatty(rid: number): boolean;
```

Reply:
Do you think the original returning a restoration function after calling `Deno.setRaw(rid)` works here? (In case I want to store the information for restoring without exposing? This would be a pure TS side change, and similar to what I did in the old PR, but without wrapping in a class)

Reply:
I don't think storing the restore state is so important. The user can do that.

Reply:
I don't think they can with current API (unless we expose a full representation `<termios>`) if we do not store for them? And there is the discrepancy between Unix and Windows here

Node does not provide the API for direct restoration either. I believe with `tty.ReadStream::setRaw(false)` libuv simply [sets back to the saved original mode](https://github.com/libuv/libuv/blob/1399d851b112ad8376ef2a46303e72fd55125439/src/unix/tty.c#L243)

Reply:
> libuv simply sets back to the saved original mode

Yes we should do that too...

Reply:
The slight annoyance is that we will need to accompany StreamResource (including `FsFile`) with such optional data. I'll see where would be the best place for me to place this piece of info.